### DISCUSSION TITLE
Simple CounterComponent breaks when used `@if` statement

### DISCUSSION THREAD

**Discussion by sanekb** [livewire/livewire#5630] [1. Help] [answered]
I was experimenting with the simple bidirectional counter and noticed a strange behavior of the @if statement. More precisely, the click-event handler is lost by divs

CounterComponent.php functionality:
```php
public $count = 1;

public function increment () {
    $this->count++;
}
public function decrement () {
    $this->count--;
}
```
and i's view:
```blade
<div>
    @if( $count > 0 )
    <div wire:click="decrement()">-</div>@endif
    <div>{{ $count }}</div>
    <div wire:click="increment()">+</div>
</div>
```
In this case, when click `-` after rerender we will see `0+`, and `+` will not work (in DevTools this div will not have an click-event handler)

but with small changes, the error already ceases to be reproduced. this examples work's fine:
```blade
@if( $count > 0 )
<div wire:click="decrement()">-</div>@endif
<span>{{ $count }}</span>
<div wire:click="increment()">+</div>
```
```blade
@if( $count > 0 )
<div wire:click="decrement()">-</div>@endif
<div id="someId">{{ $count }}</div>
<div wire:click="increment()">+</div>
```
```blade
<div>{{ $count }}</div>
@if( $count > 0 )
<div wire:click="decrement()">-</div>@endif
<div wire:click="increment()">+</div>
```

**Comment by joshhanley** [comment, marked as answer]
@sanekb you should wrap all conditionals in an element, like a div, as conditonals can cause DOM diffing issues. See here for more details https://laravel-livewire.com/docs/2.x/troubleshooting#dom-diffing-issues

Hope this helps!
